go/vt/sqlparser/ast.go

```diff
@@ -1626,6 +1637,42 @@ type PartitionValueRange struct {
 	Maxvalue bool
 }
 
+type PartitionEngine struct {
+	Storage bool
+	Equal   string
+	Name    string
+}
+
+type PartitionComment struct {
+	Equal   string
+	Comment string
+}
+
+type PartitionDataDirectory struct {
+	Equal   string
+	DataDir string
+}
+
+type PartitionIndexDirectory struct {
+	Equal    string
+	IndexDir string
+}
+
+type PartitionMaxRows struct {
+	Equal string
+	Rows  int
+}
+
+type PartitionMinRows struct {
+	Equal string
+	Rows  int
+}
+
+type PartitionTableSpace struct {
+	Equal string
+	Name  string
+}
```

Comment:
After you remove the Equal field, the structs can be unrolled into the main struct directly